src/plugin.ts

```diff
@@ -98,40 +98,32 @@ function topicsToArray(input: any, base?: string): Topic[] {
   })
 }
 
-// eslint-disable-next-line valid-jsdoc
-/**
- * find package root
- * for packages installed into node_modules this will go up directories until
- * it finds a node_modules directory with the plugin installed into it
- *
- * This is needed because of the deduping npm does
- */
-async function findRoot(name: string | undefined, root: string) {
-  // essentially just "cd .."
-  function * up(from: string) {
-    while (path.dirname(from) !== from) {
-      yield from
-      from = path.dirname(from)
-    }
+// essentially just "cd .."
+function * up(from: string) {
+  while (path.dirname(from) !== from) {
     yield from
+    from = path.dirname(from)
   }
+  yield from
+}
+
+async function findSourcesRoot(root: string) {
   for (const next of up(root)) {
-    let cur
-    if (name) {
-      cur = path.join(next, 'node_modules', name, 'package.json')
-      // eslint-disable-next-line no-await-in-loop
-      if (await exists(cur)) return path.dirname(cur)
-      try {
-        // eslint-disable-next-line no-await-in-loop
-        const pkg = await loadJSON(path.join(next, 'package.json'))
-        if (pkg.name === name) return next
-      } catch { }
-    } else {
-      cur = path.join(next, 'package.json')
-      // eslint-disable-next-line no-await-in-loop
-      if (await exists(cur)) return path.dirname(cur)
-    }
+    const cur = path.join(next, 'package.json')
+    // eslint-disable-next-line no-await-in-loop
+    if (await exists(cur)) return path.dirname(cur)
+  }
+}
+
+async function findRoot(name: string | undefined, root: string) {
+  if (name) {
+    let pkgPath
+    try {
+      pkgPath = await resolvePackage(name)
```

Comment:
I think you need to pass in `{ basedir: root }` because it may need to look outside of the current path. i.e. where user plugins are stored.

With that said, the `resolve` library isn't working how I thought it would. If I add the basedir above, it will still sometimes throw `MODULE_NOT_FOUND` even though it seems to be in the resolution path.

For example, I modified resolvePackage to take in the same options object that resolve takes and pass that to resolve.

```javascript
// name = '@salesforce/plugin-evergreen-build'
// root = '/Users/t.dvornik/.local/share/sfdx/node_modules/@salesforce/plugin-evergreen'
...
// The root may not always be a dir, so ensure it is before passing it into resolve
if (path.extname(root)) {
  root = path.dirname(root)
}
try {
  pkgPath = await util_2.resolvePackage(name, { basedir: root });
}
...
```

That package is in `/Users/t.dvornik/.local/share/sfdx/node_modules/@salesforce/plugin-evergreen-build` which should be in node module resolution but resolve isn't resolving it.

Reply:
Thanks for looking at this - this issue flew under the radar for me for a minute, but I'll have a look to see if I can fix it sometime soon. It's not high priority for me but evidently there's a few others others also waiting on a fix.